### DISCUSSION TITLE
#nullable true in 1.2.0

### DISCUSSION THREAD

**Discussion by Jappinen** [christianhelle/refitter#449] [General]
Hello,

After updating from 1.1.3 to 1.2.0 my build output lit up like a Xmas tree with nullable property warnings. Take the simples possible Petstore sample for example (on fresh .NET 8 console project with everything as default as Rider sets them up):

```
{
  "openApiPath": "https://petstore3.swagger.io/api/v3/openapi.yaml"
}
```

>petstore.g.cs(626,23): Warning CS8618 : Non-nullable property 'Type' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable.

...and continues for 27 lines. The culprit is the new "#nullable enable" line after usings and before namespace that was previously behind a flag (https://github.com/christianhelle/refitter/commit/b13015b8dd5f6345de13db2dd4067695d9bc2ff9).

After a little bit of digging I found you discussed this on https://github.com/christianhelle/refitter/pull/435 and as per the docs ([the blue box at the end of chapter](https://learn.microsoft.com/en-us/dotnet/csharp/nullable-references#nullable-contexts)) for generated code you really need to have that pragma in place to get the benefits.

In order to tell the underlying NSwag that you want nullables you need to opt-in: https://github.com/RicoSuter/NSwag/issues/2435#issuecomment-660080493

And like magic all my problems went away with the this:

```
{
  "openApiPath": "https://petstore3.swagger.io/api/v3/openapi.yaml",
  "codeGeneratorSettings": {
    "generateNullableReferenceTypes": true
  }
}
```

So I guess what I'm trying to ask is shouldn't that be the default setting from now on or am I missing something blatantly obvious?

**Comment by christianhelle** [comment]
@Jappinen Thanks for taking the time to bring this up. v1.2.0 came with a few large external contributions and I'm guessing that we missed some details during the review that are giving you this problem now

I'll investigate this and see what I can do

**Comment by christianhelle** [comment]
> After a little bit of digging I found you discussed this on #435 and as per the docs ([the blue box at the end of chapter](https://learn.microsoft.com/en-us/dotnet/csharp/nullable-references#nullable-contexts)) for generated code you really need to have that pragma in place to get the benefits.

Thanks for investigating. I looked a bit more into the [nullable references docs](https://learn.microsoft.com/en-us/dotnet/csharp/nullable-references#nullable-contexts) and can see (and tested) that if I generated

```csharp
#nullable enable annotations
```

The the warnings do not appear. According to the docs:

>_annotations_: The compiler doesn't emit warnings when code might dereference null, or when you assign a maybe-null expression to a non-nullable variable.
>- All new nullable warnings are disabled.
>- You can use the ? suffix to declare a nullable reference type.
>- Reference type variables without the ? suffix are non-nullable reference types.
>- You can use the null forgiving operator, !, but it has no effect.

I'll include a fix for this in the upcoming v1.2.1
